code/game/objects/items/tools/shovel_tools.dm

```diff
@@ -68,16 +71,17 @@
 		return
 	if(folded)
 		return
-
 	if(user.do_actions)
 		return
-
-
+	if(target.loc.z == 2 && entrenchingmode)
+		dig_trench(target, user)
 	if(isturf(target))
 		if(!dirt_amt)
 			var/turf/T = target
 			var/turfdirt = T.get_dirt_type()
 			if(turfdirt)
+				if(entrenchingmode)
+					return
```

Comment:
Why doesnt this return after the dig_trench call instead?

Reply:
done